Update upgrade docs for 1.13.2. (#15028)
hashicorp/consul: website/content/docs/upgrading/upgrade-specific.mdx

#### Service mesh deployments using auto-encrypt or auto-config

> Upgrade to **Consul version 1.13.2 or later**.

Reviewer: I think we should still qualify the conditions in which this is critical (auto-encrypt/config).
```suggestion
Upgrade to **Consul version 1.13.2 or later** if using
[auto-encrypt](/docs/agent/config/config-files#auto_encrypt) or
[auto-config](/docs/agent/config/config-files#auto_config).
```

> In Consul 1.13.1, auto-encrypt and auto-config both cause Consul

Reviewer: ```suggestion
In Consul 1.13.0 - 1.13.1, auto-encrypt and auto-config both cause Consul
```

to require TLS for gRPC communication with Envoy proxies.
In environments where Envoy proxies are not already configured

> to use TLS for gRPC, upgrading Consul 1.13.1 will cause

Reviewer: ```suggestion
to use TLS for gRPC, upgrading Consul Consul 1.13.0 - 1.13.1 will cause
```

Envoy proxies to disconnect from the control plane (Consul agents).

> If upgrading to version 1.13.2 from 1.13.1 you must enable

Reviewer: @freddygv :
- Does someone upgrading from <1.13 to 1.13.2+ need to touch `tls.grpc.use_auto_cert` at all? Or in some cases?
- Do we need to provide guidance on upgrading from 1.13.0/1 to 1.13.2? We had previously explicitly said "don't upgrade to 1.13 if using auto-config/encrypt", so I worry this paragraph is unnecessary detail that makes the guidance seem more complex.

Author: The motivation for adding this was a user incident where they upgraded from 1.13.1 to 1.13.2, and not using this new flag broke their setup. This set of circumstances is likely quite rare, but I erred on the side of documenting it because of someone having run into it.

Though this is actually relevant to someone upgrading from <= 1.13.1. The restriction about upgrading from 1.13.1 can be relaxed.

#### Service mesh deployments using auto-encrypt or auto-config

Upgrade to **Consul version 1.13.2 or later** if using
[auto-encrypt](/docs/agent/config/config-files#auto_encrypt) or
[auto-config](/docs/agent/config/config-files#auto_config).

In Consul 1.13.0 - 1.13.1, auto-encrypt and auto-config both cause Consul
to require TLS for gRPC communication with Envoy proxies.
In environments where Envoy proxies are not already configured

> to use TLS for gRPC, upgrading Consul Consul 1.13.0 - 1.13.1 will cause

Reviewer: ```suggestion
to use TLS for gRPC, upgrading to Consul 1.13.0 - 1.13.1 will cause
```

Envoy proxies to disconnect from the control plane (Consul agents).

> If upgrading to version 1.13.2 you must also enable

Reviewer: ```suggestion
If upgrading to version 1.13.2 or later, you must enable
```

[tls.grpc.use_auto_cert](/docs/agent/config/config-files#use_auto_cert)

> if you currently use the Connect CA and auto-encrypt certs for TLS

Reviewer: Doesn't the "if you currently use the Connect CA" part apply to anyone using Consul service mesh? (Why is that part of the qualifier needed?)

Author: The idea is that non-mutual TLS requires: servers to provide a server certificate, and clients to trust the CA that signed the server certificate. 

So "Connect CA" is referring to the CA portion, and "auto-encrypt" certs to the server cert portion.

Though given that auto-encrypt requires the Connect CA, this could be rephrased as:
```suggestion
if you currently use the auto-encrypt certs as the server certs for TLS
```

What do you think?

> if you currently use the Connect CA and auto-encrypt certs for TLS
> on the gRPC port, and do not have certificates configured via
> [tls.grpc](/docs/agent/config/config-files#tls_grpc)
> or [tls.defaults](/docs/agent/config/config-files#tls_defaults).

Author: ```suggestion
if you currently use the auto-encrypt or auto-config certs as the 
server certs for TLS on the gRPC port.
```

Reviewer: Regarding "as the server certs for TLS on the gRPC port", what do we mean by "server" in this case? I don't think we mean "server agent certs", since I didn't think Consul server agents received auto-encrypt/config certs.

Reviewer: And is the tls.grpc or tls.defaults stuff still relevant? I noticed that was removed.

Author: ```suggestion
if you currently rely on Consul agents presenting the auto-encrypt or
auto-config certs as the TLS server certs for gRPC port.
```

Author: What's meant by "server" is the server side of the TLS connection, updated
